src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -2638,5 +2638,423 @@ static void verifyArray(TypeSymbol type)
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (4,35): error CS8142: Both partial method declarations, 'C.M1()' and 'C.M1()', must use the same tuple element names.
+                //     public partial (int x, int y) M1();
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentTupleNames, "M1").WithArguments("C.M1()", "C.M1()").WithLocation(4, 35));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_13()
+        {
+            var source = @"
+partial class C
+{
+    public partial (int x, long y) M1();
+    public partial (long x, int y) M1() => default; // 1
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,36): error CS8817: Both partial method declarations must have the same return type.
+                //     public partial (long x, int y) M1() => default; // 1
+                Diagnostic(ErrorCode.ERR_PartialMethodReturnTypeDifference, "M1").WithLocation(5, 36));
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_14()
+        {
+            var source = @"
+partial class C
+{
+    public partial object M1();
+    public partial dynamic M1() => null;
+    
+    public partial dynamic M2();
+    public partial object M2() => null;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_15()
+        {
+            var source = @"
+using System;
+
+partial class C
+{
+    public partial IntPtr M1();
+    public partial nint M1() => 0;
+    
+    public partial nint M2();
+    public partial IntPtr M2() => default;
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(44930, "https://github.com/dotnet/roslyn/issues/44930")]
+        public void DifferentReturnTypes_16()
+        {
+            var source = @"
+partial class C
+{
+    public partial ref int M1();
+    public partial ref readonly int M1() => throw null!; // 1
+
+    public partial ref readonly int M2();
+    public partial ref int M2() => throw null!; // 2
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,37): error CS8818: Both partial method declarations must return by reference or neither may return by reference.
```

Comment:
>Both partial method declarations must return by reference or neither may return by reference. [](start = 41, length = 93)

The message seems a little misleading since both are returning by reference.

Reply:
True however I struggled with coming up with a higher quality message "Both partial method declarations must have equivalent 'ref' or 'ref readonly' modifiers"..?

Reply:
> Maybe "Both partial method declarations must have equivalent 'ref' or 'ref readonly' modifiers"

Sounds good. Perhaps strengthen it to "... must have matching 'ref' or 'ref readonly' modifiers."

---
In reply to: [442527921](https://github.com/dotnet/roslyn/pull/45128#discussion_r442527921) [](ancestors = 442527921)